internal/cephfs/controllerserver.go

```diff
@@ -205,23 +205,7 @@ func checkValidCreateVolumeRequest(
 			return errors.New("cloning snapshot-backed volumes is currently not supported")
 		}
 	case sID != nil:
-		if vol.Size < parentVol.Size {
-			return fmt.Errorf(
-				"cannot restore from snapshot %s: volume size %d is smaller than source volume size %d",
-				sID.SnapshotID,
-				parentVol.Size,
-				vol.Size)
-		}
-
 		if vol.BackingSnapshot {
-			if vol.Size != parentVol.Size {
```

Comment:
@gman0 I am removing this check as it can cause a problem if the parent subvolume is resized after taking a snapshot, please check if am missing anything.